// @aws-sdk/client-transcribe/dist-es/commands/CreateLanguageModelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateLanguageModelRequestFilterSensitiveLog, CreateLanguageModelResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateLanguageModelCommand, serializeAws_json1_1CreateLanguageModelCommand, } from "../protocols/Aws_json1_1";
var CreateLanguageModelCommand = (function (_super) {
    __extends(CreateLanguageModelCommand, _super);
    function CreateLanguageModelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateLanguageModelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "CreateLanguageModelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateLanguageModelRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateLanguageModelResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateLanguageModelCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateLanguageModelCommand(input, context);
    };
    CreateLanguageModelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateLanguageModelCommand(output, context);
    };
    return CreateLanguageModelCommand;
}($Command));
export { CreateLanguageModelCommand };
