// @aws-sdk/client-transcribe/dist-es/commands/CreateVocabularyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateVocabularyRequestFilterSensitiveLog, CreateVocabularyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateVocabularyCommand, serializeAws_json1_1CreateVocabularyCommand, } from "../protocols/Aws_json1_1";
var CreateVocabularyCommand = (function (_super) {
    __extends(CreateVocabularyCommand, _super);
    function CreateVocabularyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateVocabularyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "CreateVocabularyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateVocabularyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateVocabularyResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateVocabularyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateVocabularyCommand(input, context);
    };
    CreateVocabularyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateVocabularyCommand(output, context);
    };
    return CreateVocabularyCommand;
}($Command));
export { CreateVocabularyCommand };
