// @aws-sdk/client-transcribe/dist-es/commands/CreateCallAnalyticsCategoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCallAnalyticsCategoryRequestFilterSensitiveLog, CreateCallAnalyticsCategoryResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateCallAnalyticsCategoryCommand, serializeAws_json1_1CreateCallAnalyticsCategoryCommand, } from "../protocols/Aws_json1_1";
var CreateCallAnalyticsCategoryCommand = (function (_super) {
    __extends(CreateCallAnalyticsCategoryCommand, _super);
    function CreateCallAnalyticsCategoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCallAnalyticsCategoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "CreateCallAnalyticsCategoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCallAnalyticsCategoryRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateCallAnalyticsCategoryResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCallAnalyticsCategoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateCallAnalyticsCategoryCommand(input, context);
    };
    CreateCallAnalyticsCategoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateCallAnalyticsCategoryCommand(output, context);
    };
    return CreateCallAnalyticsCategoryCommand;
}($Command));
export { CreateCallAnalyticsCategoryCommand };
