// @aws-sdk/client-transcribe/dist-es/commands/CreateMedicalVocabularyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateMedicalVocabularyRequestFilterSensitiveLog, CreateMedicalVocabularyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateMedicalVocabularyCommand, serializeAws_json1_1CreateMedicalVocabularyCommand, } from "../protocols/Aws_json1_1";
var CreateMedicalVocabularyCommand = (function (_super) {
    __extends(CreateMedicalVocabularyCommand, _super);
    function CreateMedicalVocabularyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateMedicalVocabularyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "CreateMedicalVocabularyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateMedicalVocabularyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateMedicalVocabularyResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateMedicalVocabularyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateMedicalVocabularyCommand(input, context);
    };
    CreateMedicalVocabularyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateMedicalVocabularyCommand(output, context);
    };
    return CreateMedicalVocabularyCommand;
}($Command));
export { CreateMedicalVocabularyCommand };
