// @aws-sdk/client-transcribe/dist-es/commands/DeleteCallAnalyticsCategoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCallAnalyticsCategoryRequestFilterSensitiveLog, DeleteCallAnalyticsCategoryResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DeleteCallAnalyticsCategoryCommand, serializeAws_json1_1DeleteCallAnalyticsCategoryCommand, } from "../protocols/Aws_json1_1";
var DeleteCallAnalyticsCategoryCommand = (function (_super) {
    __extends(DeleteCallAnalyticsCategoryCommand, _super);
    function DeleteCallAnalyticsCategoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCallAnalyticsCategoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "DeleteCallAnalyticsCategoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCallAnalyticsCategoryRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteCallAnalyticsCategoryResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCallAnalyticsCategoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteCallAnalyticsCategoryCommand(input, context);
    };
    DeleteCallAnalyticsCategoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteCallAnalyticsCategoryCommand(output, context);
    };
    return DeleteCallAnalyticsCategoryCommand;
}($Command));
export { DeleteCallAnalyticsCategoryCommand };
