// @aws-sdk/client-transcribe/dist-es/commands/DeleteVocabularyFilterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteVocabularyFilterRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_json1_1DeleteVocabularyFilterCommand, serializeAws_json1_1DeleteVocabularyFilterCommand, } from "../protocols/Aws_json1_1";
var DeleteVocabularyFilterCommand = (function (_super) {
    __extends(DeleteVocabularyFilterCommand, _super);
    function DeleteVocabularyFilterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteVocabularyFilterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "DeleteVocabularyFilterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteVocabularyFilterRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteVocabularyFilterCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteVocabularyFilterCommand(input, context);
    };
    DeleteVocabularyFilterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteVocabularyFilterCommand(output, context);
    };
    return DeleteVocabularyFilterCommand;
}($Command));
export { DeleteVocabularyFilterCommand };
