// @aws-sdk/client-transcribe/dist-es/commands/DescribeLanguageModelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeLanguageModelRequestFilterSensitiveLog, DescribeLanguageModelResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DescribeLanguageModelCommand, serializeAws_json1_1DescribeLanguageModelCommand, } from "../protocols/Aws_json1_1";
var DescribeLanguageModelCommand = (function (_super) {
    __extends(DescribeLanguageModelCommand, _super);
    function DescribeLanguageModelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeLanguageModelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "DescribeLanguageModelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeLanguageModelRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeLanguageModelResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeLanguageModelCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeLanguageModelCommand(input, context);
    };
    DescribeLanguageModelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeLanguageModelCommand(output, context);
    };
    return DescribeLanguageModelCommand;
}($Command));
export { DescribeLanguageModelCommand };
