// @aws-sdk/client-transcribe/dist-es/commands/GetCallAnalyticsJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCallAnalyticsJobRequestFilterSensitiveLog, GetCallAnalyticsJobResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetCallAnalyticsJobCommand, serializeAws_json1_1GetCallAnalyticsJobCommand, } from "../protocols/Aws_json1_1";
var GetCallAnalyticsJobCommand = (function (_super) {
    __extends(GetCallAnalyticsJobCommand, _super);
    function GetCallAnalyticsJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCallAnalyticsJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "GetCallAnalyticsJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCallAnalyticsJobRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetCallAnalyticsJobResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCallAnalyticsJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetCallAnalyticsJobCommand(input, context);
    };
    GetCallAnalyticsJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetCallAnalyticsJobCommand(output, context);
    };
    return GetCallAnalyticsJobCommand;
}($Command));
export { GetCallAnalyticsJobCommand };
