// @aws-sdk/client-transcribe/dist-es/commands/GetCallAnalyticsCategoryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCallAnalyticsCategoryRequestFilterSensitiveLog, GetCallAnalyticsCategoryResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetCallAnalyticsCategoryCommand, serializeAws_json1_1GetCallAnalyticsCategoryCommand, } from "../protocols/Aws_json1_1";
var GetCallAnalyticsCategoryCommand = (function (_super) {
    __extends(GetCallAnalyticsCategoryCommand, _super);
    function GetCallAnalyticsCategoryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCallAnalyticsCategoryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "GetCallAnalyticsCategoryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCallAnalyticsCategoryRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetCallAnalyticsCategoryResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCallAnalyticsCategoryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetCallAnalyticsCategoryCommand(input, context);
    };
    GetCallAnalyticsCategoryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetCallAnalyticsCategoryCommand(output, context);
    };
    return GetCallAnalyticsCategoryCommand;
}($Command));
export { GetCallAnalyticsCategoryCommand };
