// @aws-sdk/client-transcribe/dist-es/commands/GetMedicalTranscriptionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetMedicalTranscriptionJobRequestFilterSensitiveLog, GetMedicalTranscriptionJobResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetMedicalTranscriptionJobCommand, serializeAws_json1_1GetMedicalTranscriptionJobCommand, } from "../protocols/Aws_json1_1";
var GetMedicalTranscriptionJobCommand = (function (_super) {
    __extends(GetMedicalTranscriptionJobCommand, _super);
    function GetMedicalTranscriptionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetMedicalTranscriptionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "GetMedicalTranscriptionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetMedicalTranscriptionJobRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetMedicalTranscriptionJobResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetMedicalTranscriptionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetMedicalTranscriptionJobCommand(input, context);
    };
    GetMedicalTranscriptionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetMedicalTranscriptionJobCommand(output, context);
    };
    return GetMedicalTranscriptionJobCommand;
}($Command));
export { GetMedicalTranscriptionJobCommand };
