// @aws-sdk/client-transcribe/dist-es/commands/GetTranscriptionJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTranscriptionJobRequestFilterSensitiveLog, GetTranscriptionJobResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetTranscriptionJobCommand, serializeAws_json1_1GetTranscriptionJobCommand, } from "../protocols/Aws_json1_1";
var GetTranscriptionJobCommand = (function (_super) {
    __extends(GetTranscriptionJobCommand, _super);
    function GetTranscriptionJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTranscriptionJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "GetTranscriptionJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTranscriptionJobRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetTranscriptionJobResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTranscriptionJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetTranscriptionJobCommand(input, context);
    };
    GetTranscriptionJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetTranscriptionJobCommand(output, context);
    };
    return GetTranscriptionJobCommand;
}($Command));
export { GetTranscriptionJobCommand };
