// @aws-sdk/client-transcribe/dist-es/commands/GetVocabularyFilterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetVocabularyFilterRequestFilterSensitiveLog, GetVocabularyFilterResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetVocabularyFilterCommand, serializeAws_json1_1GetVocabularyFilterCommand, } from "../protocols/Aws_json1_1";
var GetVocabularyFilterCommand = (function (_super) {
    __extends(GetVocabularyFilterCommand, _super);
    function GetVocabularyFilterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetVocabularyFilterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "GetVocabularyFilterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetVocabularyFilterRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetVocabularyFilterResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetVocabularyFilterCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetVocabularyFilterCommand(input, context);
    };
    GetVocabularyFilterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetVocabularyFilterCommand(output, context);
    };
    return GetVocabularyFilterCommand;
}($Command));
export { GetVocabularyFilterCommand };
