// @aws-sdk/client-transcribe/dist-es/commands/ListVocabularyFiltersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListVocabularyFiltersRequestFilterSensitiveLog, ListVocabularyFiltersResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListVocabularyFiltersCommand, serializeAws_json1_1ListVocabularyFiltersCommand, } from "../protocols/Aws_json1_1";
var ListVocabularyFiltersCommand = (function (_super) {
    __extends(ListVocabularyFiltersCommand, _super);
    function ListVocabularyFiltersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListVocabularyFiltersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "ListVocabularyFiltersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListVocabularyFiltersRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListVocabularyFiltersResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListVocabularyFiltersCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListVocabularyFiltersCommand(input, context);
    };
    ListVocabularyFiltersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListVocabularyFiltersCommand(output, context);
    };
    return ListVocabularyFiltersCommand;
}($Command));
export { ListVocabularyFiltersCommand };
