// @aws-sdk/client-transcribe/dist-es/commands/ListLanguageModelsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListLanguageModelsRequestFilterSensitiveLog, ListLanguageModelsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListLanguageModelsCommand, serializeAws_json1_1ListLanguageModelsCommand, } from "../protocols/Aws_json1_1";
var ListLanguageModelsCommand = (function (_super) {
    __extends(ListLanguageModelsCommand, _super);
    function ListLanguageModelsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListLanguageModelsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "ListLanguageModelsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListLanguageModelsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListLanguageModelsResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListLanguageModelsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListLanguageModelsCommand(input, context);
    };
    ListLanguageModelsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListLanguageModelsCommand(output, context);
    };
    return ListLanguageModelsCommand;
}($Command));
export { ListLanguageModelsCommand };
