// @aws-sdk/client-transcribe/dist-es/commands/ListMedicalVocabulariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListMedicalVocabulariesRequestFilterSensitiveLog, ListMedicalVocabulariesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListMedicalVocabulariesCommand, serializeAws_json1_1ListMedicalVocabulariesCommand, } from "../protocols/Aws_json1_1";
var ListMedicalVocabulariesCommand = (function (_super) {
    __extends(ListMedicalVocabulariesCommand, _super);
    function ListMedicalVocabulariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListMedicalVocabulariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "ListMedicalVocabulariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListMedicalVocabulariesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListMedicalVocabulariesResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListMedicalVocabulariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListMedicalVocabulariesCommand(input, context);
    };
    ListMedicalVocabulariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListMedicalVocabulariesCommand(output, context);
    };
    return ListMedicalVocabulariesCommand;
}($Command));
export { ListMedicalVocabulariesCommand };
