// @aws-sdk/client-transcribe/dist-es/commands/ListVocabulariesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListVocabulariesRequestFilterSensitiveLog, ListVocabulariesResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListVocabulariesCommand, serializeAws_json1_1ListVocabulariesCommand, } from "../protocols/Aws_json1_1";
var ListVocabulariesCommand = (function (_super) {
    __extends(ListVocabulariesCommand, _super);
    function ListVocabulariesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListVocabulariesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "ListVocabulariesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListVocabulariesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListVocabulariesResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListVocabulariesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListVocabulariesCommand(input, context);
    };
    ListVocabulariesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListVocabulariesCommand(output, context);
    };
    return ListVocabulariesCommand;
}($Command));
export { ListVocabulariesCommand };
