// @aws-sdk/client-transcribe/dist-es/commands/StartCallAnalyticsJobCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartCallAnalyticsJobRequestFilterSensitiveLog, StartCallAnalyticsJobResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1StartCallAnalyticsJobCommand, serializeAws_json1_1StartCallAnalyticsJobCommand, } from "../protocols/Aws_json1_1";
var StartCallAnalyticsJobCommand = (function (_super) {
    __extends(StartCallAnalyticsJobCommand, _super);
    function StartCallAnalyticsJobCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartCallAnalyticsJobCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "TranscribeClient";
        var commandName = "StartCallAnalyticsJobCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartCallAnalyticsJobRequestFilterSensitiveLog,
            outputFilterSensitiveLog: StartCallAnalyticsJobResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartCallAnalyticsJobCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartCallAnalyticsJobCommand(input, context);
    };
    StartCallAnalyticsJobCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartCallAnalyticsJobCommand(output, context);
    };
    return StartCallAnalyticsJobCommand;
}($Command));
export { StartCallAnalyticsJobCommand };
